catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/ActiveSearch.java

```diff
@@ -0,0 +1,131 @@
+ * start time of the search {@link java.util.Date}, source being searched {@link ddf.catalog.source.Source},
+ * and uniqueID {@link java.util.UUID}
+ */
+
+public class ActiveSearch {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActiveSearch.class);
+
+    private String cqlQuery = "";
+
+    private Date startTime;
+
+    private Source source;
+
+    private UUID uniqueID = null;
+
+    private String clientInfo = "";
+
+    public String getCQL() {
+        return cqlQuery;
+    }
+
+    /**
+     * Returns a {@link Date} representing the approximate time this search started.
+     *
+     * @return {@link Date} representing the approximate time the search started. Can return {@link null}
+     */
+    public Date getStartTime() {
+        Date d = null;
+        if (startTime != null) {
+            d = (Date) startTime.clone();
+        }
+        return d;
+    }
+
+    /**
+     * Returns the source information for this {@link ActiveSearch}
+     *
+     * @return {@link Source} representing the source being queried by this {@link ActiveSearch}. Can return {@code null}
+     */
+    public Source getSource() {
+        return source;
+    }
+
+    public UUID getUniqueID() {
+        return uniqueID;
+    }
+
+    /**
+     * Returns the client information for this {@link ActiveSearch}
+     *
+     * @return {@link String} representing the client info that queried the {@link Source}.
+     */
+    public String getClientInfo() {
+        return clientInfo;
+    }
+
+    /**
+     * Constructor for an {@link ActiveSearch}.
+     *
+     * @param cqlQuery {@link String} that represents the query.
+     * @param source   {@link Source} that the search is querying.
+     * @param uniqueID {@link UUID} a unique ID that is used to uniquely identify the search.
+     */
+
+    public ActiveSearch(String cqlQuery, Source source, UUID uniqueID) {
+        if (cqlQuery == null) {
+            cqlQuery = "unknownCQLQuery";
+            LOGGER.debug("CQLQuery for new ActiveSearch is null. Being set as 'unknownCQLQuery'.");
+        }
+        this.cqlQuery = cqlQuery;
+        if (source == null) {
+            LOGGER.debug("Source for new ActiveSearch is null!");
+        }
+        this.source = source;
+        startTime = new Date();
+        if (uniqueID == null) {
+            uniqueID = java.util.UUID.randomUUID();
+            LOGGER.warn("uniqueID for new ActiveSearch is null. Setting the ID as " + uniqueID);
+        }
+        this.uniqueID = uniqueID;
+        try {
+            this.clientInfo = org.apache.shiro.SecurityUtils.getSubject()
+                    .getPrincipal()
+                    .toString();
+            if (clientInfo == null) {
+                LOGGER.debug("Could not retrieve apache principal!");
+                clientInfo = "";
+                return;
+            }
+            if (clientInfo.contains(", CN=")) {
```

Comment:
Don't do any of this, you shouldn't be digging into the principal name yourself. `SubjectUtils.getDisplayName(principal)`